enterprise/backend/src/metabase_enterprise/serialization/v2/extract.clj

```diff
@@ -152,43 +112,51 @@
-    (if analysis
-      ;; If that is non-nil, emit the report.
-      (escape-report analysis)
-      ;; If it's nil, there are no errors, and we can proceed to do the dump.
-      ;; TODO This is not handled at all, but we should be able to exclude illegal data - and it should be
-      ;; contagious. Eg. a Dashboard with an illegal Card gets excluded too.
-      (let [nodes       (set/union
-                         (u/traverse targets #(serdes/ascendants (first %) (second %)))
-                         (u/traverse targets #(serdes/descendants (first %) (second %))))
-            models      (model-set opts)
-            ;; filter the selected models based on user options
-            by-model    (-> (group-by first nodes)
-                            (select-keys models)
-                            (update-vals #(set (map second %))))
-            extract-ids (fn [[model ids]]
-                          (serdes/extract-all model (merge opts {:where [:in :id ids]})))]
+  (let [inner-targets (if (seq targets)
+                        (mapv parse-target targets)
+                        (mapv vector (repeat "Collection") (collection-set-for-user user-id)))
+        nodes         (set/union
+                       (u/traverse inner-targets #(serdes/ascendants (first %) (second %)))
+                       (u/traverse inner-targets #(serdes/descendants (first %) (second %))))
+        reasons       (u/index-by #(take 2 %) #(first (nnext %)) nodes)
+        by-model      (u/group-by first second nodes)
+        escaped       (escape-analysis by-model reasons)]
+    (if (seq escaped)
+      (log-escape-report! escaped)
+      (let [models         (model-set opts)
+            coll-set       (get by-model "Collection")
+            by-model       (select-keys by-model models)
+            extract-by-ids (fn [[model ids]]
+                             (serdes/extract-all model (merge opts {:collection-set coll-set
+                                                                    :where          [:in :id ids]})))
+            extract-all    (fn [model]
+                             (serdes/extract-all model (assoc opts :collection-set coll-set)))]
         (eduction cat
-                  [(eduction (map extract-ids) cat by-model)
+                  [(if (seq targets)
+                     (eduction (map extract-by-ids) cat by-model)
+                     (eduction (map extract-all) cat (set/intersection (set serdes.models/content) models)))
                    ;; extract all non-content entities like data model and settings if necessary
                    (eduction (map #(serdes/extract-all % opts)) cat (remove (set serdes.models/content) models))])))))
 
 (defn extract
   "Returns a reducible stream of entities to serialize"
-  [{:keys [targets] :as opts}]
+  [opts]
   (serdes.backfill/backfill-ids!)
-  (if (seq targets)
-    (extract-subtrees opts)
-    (extract-metabase opts)))
+  (extract-subtrees opts))
+
+(comment
```

Comment:
Please **leave this in!** I love these kinds of `comment` blocks and my personal code is full of them. I'm in favour of adopting that practice at Metabase.